app/assets/stylesheets/header.scss

```diff
@@ -118,31 +118,107 @@
       }
     }
   }
-  [class^="entypo-"], [class*="entypo-"] {
-    color: inherit;
-    font-size: $font-size-h3;
-    vertical-align: middle;
-  }
-  .nav-badge {
-    margin-bottom: -2px;
-    .badge {
-      position: absolute;
-      right: 10px;
-      top: 10px;
-    }
-  }
 
-  #user_menu {
+  .user-avatar {
+    height: $navbar-height;
+    margin-bottom: -$navbar-padding-vertical;
+    margin-right: 10px;
+    margin-top: -$navbar-padding-vertical;
+    padding: ($navbar-height - 30px) / 2 0;
+
     .avatar {
       height: 30px;
       width: 30px;
     }
-    .user-avatar {
-      height: $navbar-height;
-      padding: ($navbar-height - 30px)/2 0;
-      margin-bottom: -$navbar-padding-vertical;
-      margin-top: -$navbar-padding-vertical;
-      margin-right: 10px;
+  }
+
+  @media (max-width: $grid-float-breakpoint-max) {
+
+    .navbar-nav.hidden-xs {
+      margin: 0;
+      margin-top: 6px;
+    }
+
+    .nav-badge {
+      color: $navbar-inverse-link-color;
+      &:hover { color: $navbar-inverse-link-hover-color; }
+      &:hover,
+      &:focus {
+        background-color: transparent;
+      }
+    }
+
+    .form-group,
+    .twitter-typeahead {
+      margin-bottom: 0;
+
+      &,
+      input { width: 100%; }
+    }
+
+    .nav-badges .dropdown-menu {
+      width: 300px;
+    }
+
+    .user-menu-dropdown {
+      background-color: transparent;
+      border: 0;
+      box-shadow: none;
+      display: block;
+      margin-top: -8px; // To compensate parent ul margin
+      padding: 0;
+      position: static;
+      width: 100%;
+
+      a {
+        color: #9d9d9d;
```

Comment:
Color literals like `#9d9d9d` should only be used in variable declarations; they should be referred to via variable everywhere else.